caching/invalidation.py

```diff
@@ -37,22 +39,22 @@
 
 def make_key(k, with_locale=True):
     """Generate the full key for ``k``, with a prefix."""
-    key = encoding.smart_str('%s:%s' % (CACHE_PREFIX, k))
+    key = encoding.smart_text('%s:%s' % (CACHE_PREFIX, k))
     if with_locale:
-        key += encoding.smart_str(translation.get_language())
+        key += encoding.smart_text(translation.get_language())
     # memcached keys must be < 250 bytes and w/o whitespace, but it's nice
     # to see the keys when using locmem.
-    return hashlib.md5(key).hexdigest()
+    return hashlib.md5(encoding.smart_bytes(key)).hexdigest()
 
 
 def flush_key(obj):
     """We put flush lists in the flush: namespace."""
-    key = obj if isinstance(obj, basestring) else obj.cache_key
+    key = obj if isinstance(obj, basestring_) else obj.cache_key
```

Comment:
Django incorporates a copy of the `six` library, so we can do `from django.utils import six` and then: `isinstance(obj, six.string_types)`. https://docs.djangoproject.com/en/1.8/topics/python3/#writing-compatible-code-with-six